run.sh

```diff
@@ -117,16 +117,16 @@ docker run --name client -h client --cap-add=NET_ADMIN -t siff /bin/bash -c "
     route add default gw client-router
     iptables -A OUTPUT -j NFQUEUE --queue-num 0
     iptables -A INPUT -j NFQUEUE --queue-num 1
-    /go/bin/cs4516project -mode client" >> client.log 2>&1 &
-
-docker run --name attacker -h attacker --cap-add=NET_ADMIN -t siff /bin/bash -c "
-    echo -e '$HOSTS' > /etc/hosts
-    ip addr flush dev eth0
-    ip addr add $ATTACKER dev eth0
-    route add -host legacy-router/32 eth0
-    route add -host server/32 gw legacy-router
-    route add default gw legacy-router
-    iptables -A OUTPUT -j NFQUEUE --queue-num 0
-    /go/bin/cs4516project -mode attacker" >> attacker.log 2>&1
+    /go/bin/cs4516project -mode client" >> client.log 2>&1
+
+#docker run --name attacker -h attacker --cap-add=NET_ADMIN -t siff /bin/bash -c "
```

Comment:
Hm. Well it works for every other host, so I'm not sure. We might be just
dropping packets with the evil bit set.

On Mon, Dec 14, 2015 at 6:58 PM, Alec Thompson <notifications@github.com>
wrote:

> In run.sh
> <https://github.com/ThomasJClark/cs4516project/pull/30#discussion_r47578788>
> :
>
> > @@ -117,16 +117,16 @@ docker run --name client -h client --cap-add=NET_ADMIN -t siff /bin/bash -c "
> >      route add default gw client-router
> >      iptables -A OUTPUT -j NFQUEUE --queue-num 0
> >      iptables -A INPUT -j NFQUEUE --queue-num 1
> > -    /go/bin/cs4516project -mode client" >> client.log 2>&1 &
> > -
> > -docker run --name attacker -h attacker --cap-add=NET_ADMIN -t siff /bin/bash -c "
> > -    echo -e '$HOSTS' > /etc/hosts
> > -    ip addr flush dev eth0
> > -    ip addr add $ATTACKER dev eth0
> > -    route add -host legacy-router/32 eth0
> > -    route add -host server/32 gw legacy-router
> > -    route add default gw legacy-router
> > -    iptables -A OUTPUT -j NFQUEUE --queue-num 0
> > -    /go/bin/cs4516project -mode attacker" >> attacker.log 2>&1
> > +    /go/bin/cs4516project -mode client" >> client.log 2>&1
> > +
> > +#docker run --name attacker -h attacker --cap-add=NET_ADMIN -t siff /bin/bash -c "
>
> I don't think it works, since running it doesn't produce any logs except
> in attacker.log, and it says that there's no route to host when it tries to
> send an HTTP request to the server.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ThomasJClark/cs4516project/pull/30/files#r47578788>.
>



-- 
Brett Ammeson